kubespawner/spawner.py

```diff
@@ -1492,10 +1492,10 @@ def is_pod_running(self, pod):
         # FIXME: Validate if this is really the best way
         is_running = (
             pod is not None and
-            pod.status.phase == 'Running' and
-            pod.status.pod_ip is not None and
-            pod.metadata.deletion_timestamp is None and
-            all([cs.ready for cs in pod.status.container_statuses])
+            pod["status"]["phase"] == 'Running' and
+            pod["status"]["podIP"] is not None and
+            "deletionTimestamp" not in pod["metadata"] and
```

Comment:
Hmm, maybe this should be:

`not pod["metadata"].get("deletionTimestamp")`

Because `deletionTimestamp` not being in the dict vs being in the dict with a None value are different things, unless that's not possible with how the kube API works?

Reply:
`deletionTimestamp` is only set by the server when a deletion is requested. It shouldn't exist at any other time.